### PR TITLE
chores: npm extension: add shell option to execFile in windows env

### LINKED ISSUE #211914: Spurious ${LOCALAPPDATA} folder created when hovering a dependency name in package.json
Type: <b>Bug</b>

1. Open any folder with a `package.json` file.
2. Hover any dependency name in `package.json` to display the package's details.
3. An extra folder named `${LOCALAPPDATA}` appears in the same directory where the `package.json` file resides.

image

It appears from the logs in that folder that `vscode` launches `npm view` to get the package details:

image

So I tried the same command from the shell:

```bash
npm view --json -- eslint description dist-tags.latest homepage version time
```

The extra folder was not created.

VS Code version: Code 1.89.0 (b58957e67ee1e712cebf466b995adf4c5307b2bd, 2024-05-01T02:09:22.859Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i7-12700K (20 x 3610)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.75GB (17.10GB free)|
|Process Argv|--disable-extensions . --crash-reporter-id 365a2674-a228-4abd-82c2-408321056497|
|Screen Reader|no|
|VM|0%|
</details>Extensions disabled

### PR DESCRIPTION
**Pull request by frankli0324**
resolves CVE-2024-27980

fixes: https://github.com/microsoft/vscode/issues/211914

### CODE REVIEW

File: extensions/npm/src/features/packageJSONContribution.ts [hunk 1]
```diff
@@ -288,11 +288,12 @@ export class PackageJSONContribution implements IJSONContribution {
 			const args = ['view', '--json', '--', pack, 'description', 'dist-tags.latest', 'homepage', 'version', 'time'];
 			const cwd = resource && resource.scheme === 'file' ? dirname(resource.fsPath) : undefined;
 
+			let env = Object.assign({}, process.env);
```
Comment (Septh) [inline extensions/npm/src/features/packageJSONContribution.ts:291]: I would go with a one-liner:

```javascript
const env = { ...process.env, COREPACK_ENABLE_AUTO_PIN: "0", COREPACK_ENABLE_PROJECT_SPEC: "0" }
```

but yeah, this does the trick.

**Comment by Septh** [comment]
I believe [CVE-2024-27980](https://nodejs.org/en/blog/vulnerability/april-2024-security-releases-2) should be considered here:

```javascript
    // corepack npm wrapper would automatically update package.json. disable that behavior.
    // COREPACK_ENABLE_AUTO_PIN disables the package.json overwrite, and
    // COREPACK_ENABLE_PROJECT_SPEC makes the npm view command succeed
    //   even if packageManager specified a package manager other than npm.
    const env = { ...process.env, COREPACK_ENABLE_AUTO_PIN: "0", COREPACK_ENABLE_PROJECT_SPEC: "0" }
    const shell = process.platform === 'win32'
    execFile(npmCommandPath, args, { cwd, env, shell }, (error, stdout) => {
```

**Comment by frankli0324** [comment]
> I believe [CVE-2024-27980](https://nodejs.org/en/blog/vulnerability/april-2024-security-releases-2) should be considered here:

sry for not being familiar with windows but under what circumstances would npm be a cmd/bat script instead of an executable?

**Comment by meganrogge** [comment]
Closing as the linked bug has been fixed

**Pull request closed without merging**